include/zephyr/bluetooth/mesh/cdb.h

```diff
@@ -247,6 +247,9 @@ void bt_mesh_cdb_node_foreach(bt_mesh_cdb_node_func_t func, void *user_data);
  *
  *  Allocate a new subnet in the CDB.
  *
+ *  @note The provided net_idx must be unique, since multiple keys for the same index
```

Comment:
Hm. This is possible to delete existing entry with net_idx1 and reallocate it again. In this case, you do not need a unique index.